pandas/core/ops/array_ops.py

```diff
@@ -265,8 +268,7 @@ def comparison_op(left: ArrayLike, right: Any, op) -> ArrayLike:
         with warnings.catch_warnings():
             # suppress warnings from numpy about element-wise comparison
             warnings.simplefilter("ignore", DeprecationWarning)
-            with np.errstate(all="ignore"):
```

Comment:
should the catch_warnings here be done at a higher level too?

Reply:
Possibly, but that might need to be done more targeted for the specific type of op (since we don't do anything similar in arithmetic_op), and thus this will be more complicated. So would like to start with `np.errstate`.